api/src/controllers/AuthenticationController.ts

```diff
@@ -31,10 +32,25 @@ export default class AuthenticationContoller extends ErrorController {
         throw ErrorController.ForbiddenError();
       }
 
+      if (userDetails.mfaSecret !== null) {
+        if (data.mfaCode === undefined) {
+          Logger.error(`Missing mfa code for user '${data.username}' with MFA enabled.`);
+          throw ErrorController.ForbiddenError();
+        }
+        const isValidMfa = await this.authService.mfaVerification(
+          userDetails.mfaSecret,
+          data.mfaCode,
+        );
+        if (!isValidMfa) {
+          Logger.error(`Invalid MFA code provided for user '${data.username}`);
+          throw ErrorController.ForbiddenError();
+        }
+      }
+
       Logger.info('User signed in successfully');
       res.status(200).json({
         idToken: this.authService.generateIdToken(userDetails),
-        accessToken: this.authService.generateAccessToken(userDetails.scope),
+        accessToken: this.authService.generateAccessToken(userDetails.scope, userDetails.username),
```

Comment:
Why are we passing the username in the access token?

Reply:
To provide the `sub` value within `generateClaims` to be representable as the subject, which was updated to allow the user to be identified within the `initiMfa` controller in order to retrieve the user from the database. Since an alternative would be to provide the `username` as an additional field, in which that has no security enforcing that the owner of the access token is matching the requested username - allowing anyone to setup MFA for any account. Another option would be to require the endpoint to require a refresh token instead of access, since that has the username as part of it - but didn't quite seem right to require a refresh token for a non-refresh purpose.